app.py

```diff
@@ -31,46 +30,176 @@ def init_app(self):
+
+        # Check that we don't have a file path.
+        if os.path.isfile(path):
+            raise ValueError(
+                "The path should be a folder not be a file path. "
+                "If you want to open a folder containing a file use _launch_filemanager_for_file. Path: %s" % path)
+
+        # get the setting
         system = sys.platform
-        
+
         # run the app
-        if system == "linux2":
+        if system.startswith("linux2"):
             cmd = 'xdg-open "%s"' % path
         elif system == "darwin":
             cmd = 'open "%s"' % path
         elif system == "win32":
             cmd = 'cmd.exe /C start "Folder" "%s"' % path
         else:
             raise Exception("Platform '%s' is not supported." % system)
-        
+
         self.log_debug("Executing command '%s'" % cmd)
         exit_code = os.system(cmd)
         if exit_code != 0:
             self.log_error("Failed to launch '%s'!" % cmd)
 
-    
+    def _launch_filemanager_for_file(self, path):
+        """
+        This method will take a path to a file and open it in the OS's default file manager.
+        This is only used for files, for folders use _launch_filemanager_for_folder.
+
+        :param path: A file path
+        :return: None
+        """
+        self.log_debug("Launching file system viewer for file %s" % path)
+
+        if not os.path.isfile(path):
+            raise ValueError(
+                "The path should be a file path not be a folder path. "
+                "If you want to open a folder use _launch_filemanager_for_folder. Path: %s" % path)
+
+        # get the setting
+        system = sys.platform
+
+        # build a command that will open and ideally select the file in the OS's default file manager.
+        if system.startswith("linux"):
+            # Can't find a reliable way to open a file browser and select a file on linux,
+            # so if we get passed a file path, only open up the folder.
+            cmd = 'xdg-open "%s"' % os.path.dirname(path)
+        elif system == "darwin":
+            cmd = 'open -R "%s"' % path
+        elif system == "win32":
+            cmd = 'explorer /select,"%s"' % path
+        else:
+            raise Exception("Platform '%s' is not supported." % system)
+
+        self.log_debug("Executing command '%s'" % cmd)
+        exit_code = os.system(cmd)
+        if exit_code != 0:
+            self.log_error("Failed to launch '%s'!" % cmd)
+
+    def _get_published_file_path(self, entity_id):
+        """
+        Will attempt to return back a resolved path for a PublishedFile entity
+        :param entity_id: int for the PublishedFile ID.
+        :return: str path or None depending on whether it successfully resolves the path.
+        """
+
+        # Find the publish so that we can get the path back.
+        published_file = self.sgtk.shotgun.find_one("PublishedFile", [["id", "is", entity_id]], ["code", "path"])
```

Comment:
nit: code seems unused.

Reply:
It's not clear in the GitHub review which line you are referring to, but I can't see any code that is not in use. Can you clarify which line? Cheers

Reply:
sorry - the `code` field you are requesting from shotgun seems unused.

Reply:
Ah gotcha, heheh, I thought you meant some python code, not the code field its self. yep cool, I'll address that.

Reply:
There is a bug with the `resolve_publish_path` which means that it requires the published_file to have the `code` key present, in certain situations. That bug is logged, so I'll leave this as it is a make a comment explaining, why the code is being gathered. @jfboismenu